resolver/addressable_resolver.go

```diff
@@ -65,20 +64,28 @@ func NewURIResolver(ctx context.Context, callback func(types.NamespacedName)) *U
 
 // URIFromDestination resolves a Destination into a URI string.
 func (r *URIResolver) URIFromDestination(dest apisv1alpha1.Destination, parent interface{}) (string, error) {
-	// Prefer resolved object reference + path, then try URI + path, honoring the Destination documentation
-	if dest.ObjectReference != nil {
-		url, err := r.URIFromObjectReference(dest.ObjectReference, parent)
+	if dest.Ref != nil {
+		url, err := r.URIFromObjectReference(dest.Ref, parent)
 		if err != nil {
 			return "", err
 		}
-		return extendPath(url.DeepCopy(), dest.Path).String(), nil
+		if dest.URI != nil {
+			if dest.URI.URL().IsAbs() {
+				return "", fmt.Errorf("absolute URI is not allowed when Ref exists")
+			}
+			return url.URL().ResolveReference(dest.URI.URL()).String(), nil
+		}
+		return url.URL().String(), nil
 	}
 
 	if dest.URI != nil {
-		return extendPath(dest.URI.DeepCopy(), dest.Path).String(), nil
+		if !dest.URI.URL().IsAbs() || dest.URI.Host == "" {
+			return "", fmt.Errorf("invalid URI: %v", dest.URI.String())
```

Comment:
The error string should be `URI is not absolute: %v` to tell the user exactly what's invalid about it.